src/nodedef.cpp

```diff
@@ -809,14 +810,22 @@ void CNodeDefManager::updateTextures(IGameDef *gamedef,
 				f->drawtype = NDT_ALLFACES;
 				f->solidness = 0;
 				f->visual_solidness = 1;
+				if (f->waving == 1)
+					material_type = TILE_MATERIAL_WAVING_LEAVES;
 			} else {
-				f->drawtype = NDT_NORMAL;
-				f->solidness = 2;
-				for (u32 i = 0; i < 6; i++)
-					tiledef[i].name += std::string("^[noalpha");
+				if (plantlike_leaves) {
+					f->visual_scale = 1.1892;
+					f->drawtype = NDT_PLANTLIKE;
+					f->solidness = 0;
+					f->backface_culling = false;
+					material_type = TILE_MATERIAL_WAVING_PLANTS;
+				} else {
```

Comment:
No, l disabled it here because l think waving leaves with black look strange and you usually disable fancy leaves if you want to decrease lag.